detectors/resources-support/src/main/java/com/google/cloud/opentelemetry/detectors/AttributeKeys.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.opentelemetry.detectors;
+
+/**
+ * Contains constants that act as keys for the known attributes for {@link
+ * com.google.cloud.opentelemetry.detectors.GCPPlatformDetector.SupportedPlatform}s.
+ */
+public final class AttributeKeys {
+  // GCE Attributes
+  public static final String GCE_PROJECT_ID = AttributeKeys.PROJECT_ID;
+  public static final String GCE_AVAILABILITY_ZONE = AttributeKeys.AVAILABILITY_ZONE;
+  public static final String GCE_CLOUD_REGION = AttributeKeys.CLOUD_REGION;
+  public static final String GCE_INSTANCE_ID = AttributeKeys.INSTANCE_ID;
+  public static final String GCE_INSTANCE_NAME = AttributeKeys.INSTANCE_NAME;
+  public static final String GCE_MACHINE_TYPE = AttributeKeys.MACHINE_TYPE;
+
+  // GKE Attributes
+  public static final String GKE_POD_NAME = "gke_pod_name";
```

Comment:
The existing detector did do this. 

I saw resource attributes present in OTel semconv like `K8S_POD_NAME`, `K8S_NAMESPACE_NAME` etc which are mapped to these - why do you say we shouldn't detect these?

Reply:
If multiple detectors are used that detect a resource attribute, they can result in a collision and an error to users.  It would be preferable to have a k8s resource detector that does this for all kubernetes platforms.

Reply:
Discussed offline - will remove K8s specific attributes and match the detected attributes across platforms to be same as the go detector.